### DISCUSSION TITLE
uvicorn[standard] no longer works on arm v7 as of v0.18

### DISCUSSION THREAD

**Discussion by tedivm** [Kludex/uvicorn#1547] [Potential Issue] [answered]
v0.18 switched from watchgod to watchfiles, which also adds a dependency to the rust toolchain since watchfiles relies on the rust notify crate.

For most systems this works fine. However, right now watchfiles will not compile on ARM v7 (it requires arm64). There isn't a workaround for this other than not installing watchfiles.

The `uvicorn[standard]` tag currently includes watchfiles regardless of the system architecture, so it fails to install on arm v7 systems.

My [downstream project](https://github.com/multi-py/python-celery) has a workaround- I'm simply using a full list of packages instead of the `standard` extras. I figured I'd bring this up though in case others ran into this issue, as I'm not sure how many people build for the simpler arm platforms.

**Comment by br3ndonland** [comment, marked as answer]
> There isn't a workaround for this other than not installing watchfiles.

I definitely understand the pain point here. Adding Rust binary extensions introduces complications on various platforms and distributions, including ARM and Alpine.

Could you get them a PR with builds for ARM v7? They seem open to multi-arch builds (samuelcolvin/watchfiles#120).

> I'm simply using a full list of packages instead of the `standard` extras

There's been a lot of discussion about lumping all the optional dependencies into `uvicorn[standard]`. It's not optimal, so your approach of installing the packages separately is probably the way to go for now.

Related:

- encode/uvicorn#219
- encode/uvicorn#1274
- https://github.com/encode/uvicorn/pull/1460#issuecomment-1115422565

**Reply by tedivm** [reply]
So far watchfiles has been pretty open to PRs- they [accepted one from me](https://github.com/samuelcolvin/watchfiles/pull/153) already so I could [change the timeout period on reloads to allow Celery to finish existing jobs](https://github.com/multi-py/python-celery/blob/main/README.md#reload_sigint_timeout) before actually reloading.


It looks like they did make an arm v7 variant, but it looks like pypi is failing to notice. I'll look into it some more.
